apps/frontend/src/store/search.ts

```diff
@@ -0,0 +1,550 @@
+  @Mutation
+  CLEAR_SEVERITY() {
+    this.severityFilter = [];
+  }
+
+  // Control ID Filtering
+
+  /** Adds or replaces ID search in the search bar */
+  @Action
+  addIdSearch(id: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit('SET_SEARCH', `${this.searchTerm} AND id:${id}`);
+    } else {
+      this.context.commit('SET_SEARCH', `id:${id}`);
+    }
+  }
+
+  /** Adds control id to filter */
+  @Action
+  addIdFilter(id: string | string[]) {
+    this.context.commit('ADD_ID', id);
+  }
+
+  @Mutation
+  ADD_ID(id: string | string[]) {
+    this.controlIdSearchTerms = this.controlIdSearchTerms.concat(id);
+  }
+
+  /** Sets the control IDs filter */
+  @Mutation
+  SET_ID(ids: string[]) {
+    this.controlIdSearchTerms = ids;
+  }
+
+  /** Clears all control ID filters */
+  @Mutation
+  CLEAR_ID() {
+    this.controlIdSearchTerms = [];
+  }
+
+  // Title filtering
+
+  /** Adds or replaces title in the search bar */
+  @Action
+  addTitleSearch(title: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit(
+        'SET_SEARCH',
+        `${this.searchTerm} AND title:${title}`
+      );
+    } else {
+      this.context.commit('SET_SEARCH', `title:${title}`);
+    }
+  }
+
+  /** Adds a title filter */
+  @Action
+  addTitleFilter(title: string | string[]) {
+    this.context.commit('ADD_TITLE', title);
+  }
+
+  @Mutation
+  ADD_TITLE(title: string | string[]) {
+    this.titleSearchTerms = this.titleSearchTerms.concat(title);
+  }
+
+  /** Sets the title filters */
+  @Mutation
+  SET_TITLE(titles: string[]) {
+    this.titleSearchTerms = titles;
+  }
+
+  /** Clears all title filters */
+  @Mutation
+  CLEAR_TITLE() {
+    this.titleSearchTerms = [];
+  }
+
+  // CCI ID Filtering
+
+  /** Adds or replaces severity in the search bar */
+  @Action
+  addCCIIdSearch(cciId: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit('SET_SEARCH', `${this.searchTerm} AND nist:${cciId}`);
+    } else {
+      this.context.commit('SET_SEARCH', `nist:${cciId}`);
+    }
+  }
+
+  /** Adds CCI id to filter */
+  @Action
+  addnistIdFilter(cciId: string | string[]) {
+    this.context.commit('ADD_CCI', cciId);
+  }
+
+  @Mutation
+  ADD_CCI(cciId: string | string[]) {
+    this.nistIdFilter = this.nistIdFilter.concat(cciId);
+  }
+
+  /** Sets the CCI IDs filter */
+  @Mutation
+  SET_CCI(cciIds: string[]) {
+    this.nistIdFilter = cciIds;
+  }
+
+  /** Clears all CCI ID filters */
+  @Mutation
+  CLEAR_CCI() {
+    this.nistIdFilter = [];
+  }
+
+  // Description Filtering
+
+  /** Adds or replaces description filter in the search bar */
+  @Action
+  addDescriptionSearch(description: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit(
+        'SET_SEARCH',
+        `${this.searchTerm} AND description:${description}`
+      );
+    } else {
+      this.context.commit('SET_SEARCH', `description:${description}`);
+    }
```

Comment:
Checking to see if the current searchTerm is set seems like something that could be handled generically once and not like something that should have different code for every part of the search.